assets/blocks/editor-components/confirm-dialog/use-confirm-dialog-props.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Hook that returns the props for the component ConfirmDialog, with an additional async function that mimics the
+ * synchronous native confirm() API. Loosely inspired by react-confirm HOC.
+ *
+ * @see https://github.com/haradakunihiko/react-confirm
+ * @return {Array} The first item is the props to pass to ConfirmDialog, the second one is the async function to call to
+ * 					trigger ConfirmDialog.
+ */
+export const useConfirmDialogProps = () => {
+	const [ props, setProps ] = useState( { isOpen: false } );
+	const confirm = ( text, title, newProps = {} ) => {
```

Comment:
I'm thinking if we really need the `newProps` here. I didn't find the exact purpose in the other PR. We're not using it yet?

Also, if we use, I wonder if sending the `cancelButtonText` and `confirmButtonText` would be better. Or do you think it could vary in different calls of the `confirm`?

Reply:
Hey there @renatho! Thanks for the review!

> We're not using it yet?

Yep, we aren't using it yet. I added it after committing [9aaea09](https://github.com/Automattic/sensei/pull/5361/commits/9aaea09a1a2c61fa5cf280cfc4e3b75c6cda0e5a), which added support for the new props that you just mentioned, for setting the label used by the buttons. 

About the variation in the label of the buttons, that's....something that I don't know, to be honest. Searching on the Students Groups Projects in Sensei Pro, I can see the utility behind `confirmButtonText` for each instance where the confirmation dialog is being used there, so...I guess it could really vary between different calls of the confirm? I feel like this would depend more on WHERE the `useConfirmDialogProps` is being used (e.g.: it's being called in each component using confirm vs. it's being called in some sort of root container and passed to the child components using context or something).

Personally, I consider this more like as a helper, helping to return the appropriate props for the Confirm Dialog component. This isn't something that even Gutenberg really uses, so...adding the new parameter here doesn't look like a problem to me. Actually, thinking again, maybe even the title could be part of the newProps, to be honest.

WDYT?

Reply:
IMO, if we're just adding it now trying to foresee the future need for that, I'd remove it following the KISS principle. So when we need that, we could implement it tailored to our needs. Does it make sense?

> Actually, thinking again, maybe even the title could be part of the newProps, to be honest.

I like the idea!

Reply:
Hey there.

> IMO, if we're just adding it now trying to foresee the future need for that, I'd remove it following the KISS principle. So when we need that, we could implement it tailored to our needs. Does it make sense?

hmm, yeah, I generally agree with that idea. However, given that the Confirm Dialog component already supports those extra props AND that Sensei Pro's Confirmation Component already has that feature too, I think it's very easily something that will be used soon enough, considering design purposes.

If you feel strong about this, I can quickly remove the additional parameter, but considering the change below (putting `title` in `newProps`), and considering that we might have to change the title in the ConfirmDialog as mentioned here: 1537-gh-Automattic/sensei-pro#issuecomment-1188160003 - I'd say to just leave it in the way it is right now.

> I like the idea!

I removed the `title` to be part of the `newProps`, in the commit [8f046ba](https://github.com/Automattic/sensei/pull/5361/commits/8f046ba5e065d75d83fb7f8b748bb1b8256107ca). I also added some documentation for the confirm function in [8369814](https://github.com/Automattic/sensei/pull/5361/commits/8369814949140973b89903e4ae33b02157bc1e17) and I added some margin in the commit [13ac0cb](https://github.com/Automattic/sensei/pull/5361/commits/13ac0cb789023b448241cbf749bb548b2c30b514) to look more like the ConfirmDialog component.

WDYT?

Reply:
LGTM!